presto-iceberg/src/test/java/com/facebook/presto/iceberg/BenchmarkIcebergHadoopCatalog.java

```diff
@@ -0,0 +1,118 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableMap;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.OptionalInt;
+
+import static com.facebook.presto.iceberg.CatalogType.HADOOP;
+import static com.facebook.presto.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.apache.iceberg.FileFormat.PARQUET;
+import static org.openjdk.jmh.annotations.Mode.AverageTime;
+import static org.openjdk.jmh.annotations.Scope.Benchmark;
+
+@State(Benchmark)
+@OutputTimeUnit(MILLISECONDS)
+@BenchmarkMode(AverageTime)
+@Fork(2)
+@Warmup(iterations = 10)
+@Measurement(iterations = 10)
+public class BenchmarkIcebergHadoopCatalog
+{
+    @State(Benchmark)
+    public static class Context
+            extends IcebergBaseContext
+    {
+        private DistributedQueryRunner queryRunner;
+
+        public static DistributedQueryRunner getQueryRunner()
+                throws Exception
+        {
+            return createIcebergQueryRunner(
+                    ImmutableMap.of(),
+                    ImmutableMap.of("iceberg.catalog.type", HADOOP.name()),
+                    PARQUET,
+                    false,
+                    true,
+                    OptionalInt.of(1));
+        }
+
+        @Setup
+        public void setUp()
+                throws Exception
+        {
+            queryRunner = getQueryRunner();
+            super.setUp(queryRunner);
+        }
+
+        @TearDown
+        public void tearDown()
+        {
+            super.tearDown(queryRunner);
+        }
+    }
+
+    @Benchmark
+    public MaterializedResult benchmarkIcebergTableSelect(Context context)
+    {
+        return context.queryRunner
+                .execute("SELECT * FROM iceberg.test_db.region");
+    }
+
+    @Benchmark
+    public MaterializedResult benchmarkIcebergTableTpchQ4(Context context)
+    {
+        return context.queryRunner
```

Comment:
You're still including the queries in every benchmark. Can you try the following implementation?
```
package com.facebook.presto.iceberg;

import com.facebook.presto.testing.MaterializedResult;
import com.facebook.presto.tests.DistributedQueryRunner;
import org.openjdk.jmh.annotations.Benchmark;
import org.openjdk.jmh.annotations.Setup;
import org.openjdk.jmh.annotations.TearDown;

public abstract class BenchmarkIcebergRead
{
    protected DistributedQueryRunner queryRunner;

    public BenchmarkIceberg()
    {
        queryRunner = getQueryRunner();
    }

    @Setup
    public void setUp()
    {
        System.out.println("setUp");
        queryRunner.execute("CREATE SCHEMA iceberg.test_db");
        queryRunner.execute("CREATE TABLE iceberg.test_db.region AS SELECT * from tpch.sf1.region");
        queryRunner.execute("CREATE TABLE iceberg.test_db.orders AS SELECT * from tpch.sf1.orders");
        queryRunner.execute("CREATE TABLE iceberg.test_db.lineitem AS SELECT * from tpch.sf1.lineitem");
    }

    @TearDown
    public void tearDown()
    {
        System.out.println("tearDown");
        queryRunner.execute("DROP TABLE IF EXISTS iceberg.test_db.region");
        queryRunner.execute("DROP TABLE IF EXISTS iceberg.test_db.orders");
        queryRunner.execute("DROP TABLE IF EXISTS iceberg.test_db.lineitem");
        queryRunner.execute("DROP SCHEMA IF EXISTS iceberg.test_db");
        queryRunner.close();
        queryRunner = null;
    }

    public abstract DistributedQueryRunner getQueryRunner();

    @Benchmark
    public MaterializedResult benchmarkIcebergTableSelect()
    {
        return queryRunner
                .execute("SELECT * FROM iceberg.test_db.region");
    }

    @Benchmark
    public MaterializedResult benchmarkIcebergTableTpchQ4()
    {
        return queryRunner
                .execute("SELECT o.orderpriority, " +
                        "    count(*) AS order_count " +
                        "FROM iceberg.test_db.orders o " +
                        "WHERE o.orderdate >= DATE '1993-07-01' " +
                        "    AND o.orderdate < DATE '1993-07-01' + interval '3' month " +
                        "    AND EXISTS (SELECT * " +
                        "                FROM iceberg.test_db.lineitem l " +
                        "                WHERE l.orderkey = o.orderkey " +
                        "                    AND l.commitdate < l.receiptdate) " +
                        "GROUP  BY o.orderpriority " +
                        "ORDER  BY o.orderpriority ");
    }
}
```

```
/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.facebook.presto.iceberg;

import com.facebook.presto.tests.DistributedQueryRunner;
import com.google.common.collect.ImmutableMap;
import org.openjdk.jmh.annotations.BenchmarkMode;
import org.openjdk.jmh.annotations.Fork;
import org.openjdk.jmh.annotations.Measurement;
import org.openjdk.jmh.annotations.OutputTimeUnit;
import org.openjdk.jmh.annotations.Scope;
import org.openjdk.jmh.annotations.State;
import org.openjdk.jmh.annotations.Warmup;
import org.openjdk.jmh.runner.Runner;
import org.openjdk.jmh.runner.RunnerException;
import org.openjdk.jmh.runner.options.Options;
import org.openjdk.jmh.runner.options.OptionsBuilder;
import org.openjdk.jmh.runner.options.VerboseMode;

import java.util.OptionalInt;

import static com.facebook.presto.iceberg.CatalogType.HADOOP;
import static com.facebook.presto.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
import static java.util.concurrent.TimeUnit.MILLISECONDS;
import static org.apache.iceberg.FileFormat.PARQUET;
import static org.openjdk.jmh.annotations.Mode.AverageTime;

@State(Scope.Benchmark)
@OutputTimeUnit(MILLISECONDS)
@BenchmarkMode(AverageTime)
@Fork(1)
@Warmup(iterations = 1)
@Measurement(iterations = 1)
public class BenchmarkIcebergHadoopCatalog
        extends BenchmarkIceberg
{

    public static void main(String[] args)
            throws RunnerException
    {
        Options options = new OptionsBuilder()
                .verbosity(VerboseMode.NORMAL)
                .include(".*" + BenchmarkIcebergHadoopCatalog.class.getSimpleName() + ".*")
                .build();

        new Runner(options).run();
    }

    public DistributedQueryRunner getQueryRunner()
    {
        try {
            return createIcebergQueryRunner(
                    ImmutableMap.of(),
                    ImmutableMap.of("iceberg.catalog.type", HADOOP.name()),
                    PARQUET,
                    false,
                    true,
                    OptionalInt.of(1));
        }
        catch (Exception e) {
            e.printStackTrace();
        }

        return null;
    }
}
```

Same for the Hive one.